js/src/wp-seo-admin.js

```diff
@@ -299,6 +299,13 @@ import initializeAlgoliaSearch from "./kb-search/wp-seo-kb-search-init";
 			wpseoKillBlockingFiles( jQuery( this ).data( "nonce" ) );
 		} );
 
+		// Prevent form submission when pressing Enter on the switch-toggles.
+		jQuery( ".switch-yoast-seo input" ).on( "keydown", function( event ) {
+			if ( "keydown" === event.type && 13 === event.which ) {
```

Comment:
We are specifically in a "keydown" event listener, so kind of redundant to check for the event type again.